10up/classifai
# Added "Recommended Content" Block

### Description of the Change
This PR Adds the Initial "Recommended Content" Block backed by the Azure Personalizer. Major changes of PR are listed below.
- Added `@wordpress/scripts`
- Format js and fixed reported eslint errors.
- Integration with Azure Personalizer API.
- Added "Recommended Content" Block.

**Note:** This is yet initial implementation based on personalizer documentation and we may still have to check on the below stuff to close #69
- Include User location data in `contextFeatures` for better personalization. 
- <del>Currently, block only suggest 1 recommended content as personalizer only return 1 rewarded action. We should check on the possibility/workaround to display multiple contents.</del> (We are showing multiple items now, as decided in internal slack discussion)
- Define suggested configuration to make the plugin work more effectively (config like `model update frequency`, `reward wait time`, etc...).
- Verify content personalization with huge content and different use cases

Some Useful links for Azure Personalizer:
- [Personalizer Documentation](https://docs.microsoft.com/en-us/azure/cognitive-services/personalizer/)
- [How Personalizer works](https://docs.microsoft.com/en-us/azure/cognitive-services/personalizer/how-personalizer-works)


#### Screenshots
##### Personalizer Settings:
![image](https://user-images.githubusercontent.com/10613171/171599054-5a3bbf6d-c3c2-42b1-8717-e5bb43f31162.png)

##### Recommended Content Block:
![image](https://user-images.githubusercontent.com/10613171/181771015-162d5890-09e5-4bbe-a7d1-72483daf8d12.png)

Closes #69

### Alternate Designs
We may choose any other recommended content service, other than Azure Personalizer.

### Possible Drawbacks
Not identified any.


### Verification Process
1. Save Azure Personalizer settings. 
2. Verify it gets authenticated and saved. (wrong key we should show error on save)
3. Add "Recommended Content" Block
4. Verify Filters and various setting options work properly.
5. Verify Block content displays properly in front-end and send the reward to azure on click-on content.

### Checklist:




### Changelog Entry
> Added: Initial "Recommended Content" Block

### Credits

Props @iamdharmesh @jeffpaul @Sidsector9
